### DISCUSSION TITLE
Python wrapper

### DISCUSSION THREAD

**Discussion by nikodemas** [facebookexperimental/Robyn#397] [General]
Hi!

Is there a timeline when can the Python wrapper be expected?

It seems like a fantastic work what you have done here and I cannot wait to try it out, but sadly I am not competent in R.

Thanks a lot!

**Comment by manu-chauhan** [comment]
+1... same here. Absence of (or delay) of Python wrapper for Robyn is affecting reach or acceptance of the library. Good work, limited by R.

**Reply by gufengzhou** [reply]
Really sorry about this. We have shared in the summit 2 monthly ago that the reason of python delay is mainly due to the very active development on the R side. We've tried. It's hard to play the "catch up game" for python dev. Also the fact that Robyn uses a py package within R doesn't make things easier. We've decided to focus on R until all major features are shipped and then do the py translation

**Reply by Praesklepios** [reply]
Sad to hear but understandable. I feel other packages aiming for similiar things as Robyn like lightweight_mmm are just so far behind, so there definetly is a big hole to be filled with the wrapper. Any chance somebody can make an assumption about a release-quarter?

**Reply by bryanwhiting** [reply]
@gufengzhou can you link a YouTube to the summit? Would love to learn what the latest thinking is

**Reply by bryanwhiting** [reply]
is this it? https://www.youtube.com/watch?v=AboIsZRzlvQ

**Comment by igorskokan** [comment]
Hi Bryan

Please find here are the recordings of the 2nd Robyn Community summit(s):

* APAC+EMEA:
https://www.facebook.com/groups/robynmmm/posts/1285761782191952/
* Americas: https://www.facebook.com/groups/robynmmm/posts/1285981585503305/


Hope this helps,
Igor

**Comment by shamzos** [comment]
Hello everyone, I wanted to take a moment to express my gratitude for your work on the Robyn package. It's been very useful to me, and I'm impressed with its capabilities. I'm interested in participating in the project and perhaps expanding its potential further. Specifically, I'm thinking about either integrating it into Python or creating a Python wrapper for the existing R version, which I believe could extend its accessibility and usability. If this is something you're open to, please let me know how I can contribute and what steps you think would be beneficial for me to take. Your guidance would be greatly appreciated.

**Reply by alex-rowley** [reply]
Hello, did you get anywhere with this thought? I am quite interested in this.

**Reply by shamzos** [reply]
Hello @arowley85 ,

Thank you for showing interest in my idea. I'm glad to hear that you're intrigued by the prospect of a Python-based version of Robyn.

I wanted to share that after careful consideration, I've decided to focus my efforts on directly translating the R code into Python. I believe this approach will be more beneficial in the long run. Instead of developing a wrapper, creating a pure Python version of the package would allow for a more seamless integration into existing Python-based data science workflows, and potentially provide a more intuitive user experience.

The project is currently underway, and while it's a challenging task given the complexity of the Robyn package, I am making steady progress. I'm working diligently to ensure that the translation captures the original functionalities of the package as much as possible.
Thank you again for your interest.

**Reply by alex-rowley** [reply]
Thanks for the reply.

Let's get in touch. I had the same thought, and the same plan.

Although, unless I hear strong reasons not to, e.g. from the authors of Robyn, I would want to open source the same.

**Reply by gufengzhou** [reply]
We welcome the community to work on any enhancement of the package and open source it, incl. the Python version. 

Probably bit hard to believe but we do face capacity constraints in team to work on and  maintain the python version so far.
